### ISSUE TITLE
VSCode: Error reported when "CALL {h-schema}" is used in @Query 

### ISSUE THREAD

**Issue by awn70** [spring-projects/sts4#1395]
When I use the {h-schema} placeholder in a @Query native query with CALL for a stored procedure the following error is shown:
PostgreSQL: no viable alternative at input 'CALLh'     vscode-spring-boot(SQL_SYNTAX)

Code:
    @Modifying
    @Query( value = "CALL {h-schema}calcRequest( :i_session_id );", nativeQuery = true )
    void calcRequest( @Param( "i_session_id" ) String sessionId );

**Comment by martinlippert** [comment]
Thanks for reporting this, we will look into this to improve the validation here. In the meantime, you can switch off the query validation in the preferences (if the error is too annoying).

**Comment by shygnome** [comment]
We are also facing similar issues. We are using MySQL syntax in a `@Query` method with native query. I don't know if the issues we are facing are the same because they only show errors for specific words. Here is an example of the query snippet (We will hide the full query).

```java
@Query(value = "SELECT ..."
            + "..."
            + "WHERE lo.service_type = :serviceType "
            + "AND inv.issued_date >= :from AND inv.issued_date <= :to "
            + "..."
            + "GROUP BY inv.id", nativeQuery = true)
    List<SomeInterface> getSomeReport(String serviceType, LocalDate from, LocalDate to);
```

And the reported error,

```shell
MySQL: no viable alternative at input 'SELECT (...) WHERE lo.service_type = :serviceType AND inv.issued_date >= :from'
```

The error only highlights the `:from` word.

**Comment by BoykoAlex** [comment]
> ```java
> GROUP BY inv.id
> ```

The MySQL queries grammar we use did not allow `from` and `to` to be used as identifiers as these were reserved keywords. I've relaxed this in the grammar so now it is likely to parse the query above if you update from the snapshot repo (pushed f53f1d70d27b5eb2079927c33744e9d298d1fb0b as a fix for this) 
However, I think you can workaround this issue by replacing `from` with `fromDate` and `to ` with `toDate`.

(Not closing this issue as the problem reported in the comment above is not the same as reported initially in the description at the top)

**Comment by BoykoAlex** [comment]
Fixed via 3e44b798ef2b1dd5b2111d1eae0ca9a34c32205f
